daemon/daemon.go

```diff
@@ -920,24 +907,26 @@ func NewDaemon(c *Config) (*Daemon, error) {
 	// Then, we will calculate the IPv4 or IPv6 alloc prefix based on the IPv6
 	// or IPv4 alloc prefix, respectively, retrieved by k8s node annotations.
 	if config.Device == "undefined" {
-		nodeaddress.InitDefaultPrefix("")
+		node.InitDefaultPrefix("")
 	}
 
-	nodeaddress.SetIPv4ClusterCidrMaskSize(v4ClusterCidrMaskSize)
+	node.SetIPv4ClusterCidrMaskSize(v4ClusterCidrMaskSize)
 
 	if v4Prefix != AutoCIDR {
 		_, net, err := net.ParseCIDR(v4Prefix)
 		if err != nil {
 			log.WithError(err).WithField(logfields.V4Prefix, v4Prefix).Fatal("Invalid IPv4 allocation prefix")
 		}
-		nodeaddress.SetIPv4AllocRange(net)
+		node.SetIPv4AllocRange(net)
 	}
 
 	if v4ServicePrefix != AutoCIDR {
-		_, _, err := net.ParseCIDR(v4ServicePrefix)
+		_, ipnet, err := net.ParseCIDR(v4ServicePrefix)
 		if err != nil {
 			log.WithError(err).WithField(logfields.V4Prefix, v4ServicePrefix).Fatal("Invalid IPv4 service prefix")
 		}
+
+		node.AddAuxPrefix(ipnet)
```

Comment:
This should be the else case for if err != nil since we don't exit/return on error.

Reply:
oh, nevermind, it's a fatal